### ISSUE TITLE
Experimental optional fields

### ISSUE THREAD

**Issue by tampler** [thesamet/sbt-protoc#201]
Hello

I'm trying to implement the following message:
```
syntax = "proto3";

message UserResp {
  Status status = 1;
  optional User user = 2;
}
```
This fails to compile with the following: 
`[error] user.proto: This file contains proto3 optional fields, but --experimental_allow_proto3_optional was not set`

I then added the following:

`PB.protocOptions in Compile := Seq("--experimental_allow_proto3_optional")`

Which yields in 
`[error] user.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-jvm_0 hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.--jvm_0_out: `

My questions are:
1. Would you consider adding this into your project ?
2. Would you like to have a contribution for this ?

Currently, I may replace `optional User` with `oneof`, but this seems a good boilerplate eliminator and has been graduated by Google into prod

Sincerely,
Boris

**Comment by thesamet** [comment]
Hi Boris, this option is supported since 0.11.0-M3 (see https://github.com/scalapb/ScalaPB/blob/master/CHANGELOG.md#0110-m3). Can you try with this version?

**Comment by tampler** [comment]
Hi Nadav

Yes, I've been trying exactly with that option, but it seems not to be working for me:
```bash
sbt:client> protocVersion
[info] 3.14.0

sbt:client> protocOptions
[info] * --experimental_allow_proto3_optional

sbt:client> protocGenerate
[error] user.proto: is a proto3 file that contains optional fields, but code generator protoc-gen-jvm_0 hasn't been updated to support optional fields in proto3. Please ask the owner of this code generator to support proto3 optional.--jvm_0_out: 
```
My setup: 
Ubuntu 20.04.1
sbt 3.13
OpenJdk 11
sbt-protoc:  1.0.0-RC4
grpc: 0.4.1+9-4c6e96ef-SNAPSHOT

**Comment by thesamet** [comment]
Can you confirm that you're also on *ScalaPB* 0.11.0-M3 or later? If so, can you provide a minimal test project since I wasn't able to reproduce this problem.

Also, note that adding an `optional` modifier to a message field in proto3 has no effect - presence is tracked for message fields by default regardless of this modifier. The `optional` modifier makes a difference only for scalar fields.

**Comment by thesamet** [comment]
Closing since proto3-optional is available in 0.11.x
